scripts/tests/run_tv_casting_test.py

```diff
@@ -0,0 +1,178 @@
+
+                    vendor_id = next_line.split(":")[-1].strip()
+                    vendor_id = vendor_id.replace('\x1b[0m', '')
+
+                elif "Product ID:" in next_line:
+                    print(next_line)
+
+                    product_id = next_line.split(":")[-1].strip()
+                    product_id = product_id.replace('\x1b[0m', '')
+
+                elif "Device Type:" in next_line:
+                    print(next_line)
+
+                    device_type = next_line.split(":")[-1].strip()
+                    device_type = device_type.replace('\x1b[0m', '')
+
+                elif "commissioner(s) discovered" in next_line:
+                    break
+
+                # If the next line starts with "Discovered Commissioner", break the loop
+                if "Discovered Commissioner" in next_line:
+                    break
+
+            # Append the extracted information to the devices list
+            discovered_commissioners.append({
+                "discovered_commissioner": commissioner,
+                "vendor_id": vendor_id,
+                "product_id": product_id,
+                "device_type": device_type
+            })
+
+    # If the list of discovered commissioners is empty and we didn't find the "No commissioner discovered" string,
+    # then something went wrong. Exit on error.
+    if len(discovered_commissioners) == 0:
+        print('Discovery failed! No commissioner(s) discovered! The list of discovered commissioner(s) is empty!')
+
+        dump_logs_to_console('./scripts/tests/Linux-tv-casting-app-logs.txt')
+
+        remove_log_file('./scripts/tests/Linux-tv-casting-app-logs.txt')
+        remove_log_file('./scripts/tests/Linux-tv-app-logs.txt')
+
+        sys.exit(1)
+
+
+# Test if the Linux tv-casting-app is able to discover the Linux tv-app.
+# The Linux tv-casting-app and the tv-app will be run in separate processes.
+# Their corresponding output will be written to their respective log files.
+# The log file of the tv-casting-app will be parsed for strings of interest
+# which will be printed to the console.
+def test_discovery_fn():
+    with open('./scripts/tests/Linux-tv-app-logs.txt', 'w') as fd1, open('./scripts/tests/Linux-tv-casting-app-logs.txt', 'w') as fd2:
+
+        # Run the Linux tv-app and write the output to file
+        tv_app_rel_path = 'out/tv-app/chip-tv-app'
+        tv_app_abs_path = os.path.abspath(tv_app_rel_path)
+        subprocess.Popen(tv_app_abs_path, stdout=fd1, stderr=subprocess.PIPE, text=True)
+
+        time.sleep(5)
+
+        # Run the Linux tv-casting-app and write the output to file
+        tv_casting_app_rel_path = 'out/tv-casting-app/chip-tv-casting-app'
+        tv_casting_app_abs_path = os.path.abspath(tv_casting_app_rel_path)
+        subprocess.Popen(tv_casting_app_abs_path, stdout=fd2, stderr=subprocess.PIPE, text=True)
+
+    # Wait for the processes to finish writing before attempting to read
+    time.sleep(15)
```

Comment:
Is it possible to multitask i.e. parse the output as it is generated and fail-fast in cases there is a failure?

Reply:
Sure, I can update to that approach.

Reply:
Are we still waiting for time.sleep(PARSE\_INTERVAL) i.e. 15 seconds before parsing the logs?

Reply:
Yes.

Reply:
`time.sleep(PARSE_INTERVAL)` has now been removed as we will be parsing the output as we receive them in real-time.